cl/lib/email_backends.py

```diff
@@ -45,6 +135,9 @@ def send_messages(
         base_backend = settings.BASE_BACKEND
         connection = get_connection(base_backend)
         connection.open()
+        r = make_redis_interface("CACHE")
+        # check the emergency brake before entering the loop
+        check_emergency_brake(r)
```

Comment:
When checking the number of times `check_emergency_brake(r)` is called, I believe this can be minimized.

In the process of sending mass emails, this introduces only one extra call, but if we're sending emails independently, it results in two calls to `check_emergency_brake(r)` for every email sent.

If I'm not wrong we could call `check_emergency_brake(r)` just once, right before this line: `stored_id = store_message(email_message)`.

This way, we verify the brake before storing or sending the email.

Reply:
got it! thanks for your suggestion.